### DISCUSSION TITLE
Can't see ROS2 raw messages, waits indefinitely.

### DISCUSSION THREAD

**Discussion by simheo** [foxglove/community#75] [Q&A]
Hi,
I'm just starting to use Foxglove to visualise the topics from my robot.

When I connect Foxglove to ROS2 using the correct ROS_DOMAIN_ID and add a Raw Messages panel, I can see all the topics from my robot listed in a dropdown.
However, when I try to visualise the topic, I got the info "waiting for next message...".
There is no custom messages used in the topic and If I do a _ros2 topic echo_ I can see that there are messages published.

I'm using ROS2 Foxy on Ubuntu 20.04 and the desktop app of Foxglove Studio.

Any help would be appreciated.
Thanks!

**Comment by jhurliman** [comment]
@simheo is the ROS2 publisher running on the same computer that you are running Studio on, or a separate computer/VM/container/etc?

**Comment by jtbandes** [comment]
Hi! Sorry you're having trouble –
- What is the datatype of the topic(s) that don't work?
- Are you running Foxglove Studio on the same computer as ROS 2 (not using a VM, etc.)?
- If you open the developer console in Foxglove Studio, or look at the console output from the command line, do you see any warnings or errors?

**Comment by simheo** [comment]
Thanks for replying @jhurliman @jtbandes.

To answer your questions:

- the ROS2 publisher is running on the same computer as Studio,
- the data published is a JointState message from sensor_msgs,

My publisher is defined like this:
` self.joint_state_publisher = self.create_publisher(
            msg_type=JointState,
            topic='joint_states',
            qos_profile=5,)`

- I think I opened the Developer Tools console (I followed the indications from [here](https://foxglove.dev/docs/panels/studio-logs) but I don't see any errors or warnings appearing.
